src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -143,70 +144,73 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         return true;
     }
 
-    static List<InclusionExclusionPattern> findTestSplits(Parallelism parallelism, Run<?,?> build, TaskListener listener, boolean generateInclusions) {
+    static List<InclusionExclusionPattern> findTestSplits(Parallelism parallelism, Run<?,?> build, TaskListener listener,
+                                                          boolean generateInclusions,
+                                                          @CheckForNull final PreviousTestResultLookup lookup) {
         TestResult tr = findPreviousTestResult(build, listener);
         if (tr == null) {
             listener.getLogger().println("No record available, so executing everything in one place");
             return Collections.singletonList(new InclusionExclusionPattern(Collections.<String>emptyList(), false));
-        } else {
-
-            Map<String/*fully qualified class name*/, TestClass> data = new TreeMap<>();
-            collect(tr, data);
-
-            // sort in the descending order of the duration
-            List<TestClass> sorted = new ArrayList<>(data.values());
-            Collections.sort(sorted);
-
-            // degree of the parallelism. we need minimum 1
-            final int n = Math.max(1, parallelism.calculate(sorted));
+        } else if (lookup != null) {
+            tr = lookup.lookupTestResult(tr);
```

Comment:
The API would seem to make more sense if it were called on `(build, listener)` in place of the default call to `findPreviousTestResult`.

Reply:
Thought about that, not yet sure what felt best. I’ll experiment some more.